jerry-core/parser/js/js-parser.c

```diff
@@ -2533,6 +2533,13 @@ parser_parse_class_fields (parser_context_t *context_p) /**< context */
     if (class_field_type & PARSER_CLASS_FIELD_NORMAL)
     {
       parser_emit_cbc_literal (context_p, CBC_ASSIGN_PROP_THIS_LITERAL, literal_index);
+      JERRY_ASSERT (context_p->stack_depth <= 2);
```

Comment:
Is this the correct assertion? It must be only me, but I cannot trivially see why this assert ensures `context_p->stack_limit <= PARSER_MAXIMUM_STACK_LIMIT` (which is the invariant that needs to hold, if I'm right). Wouldn't `JERRY_ASSERT (context_p->stack_limit <= PARSER_MAXIMUM_STACK_LIMIT);` be the right thing to add _after_ `context_p->stack_limit++;` ?

Reply:
The `stack limit` is only increased by one if the actual `stack_depth` is equal to the `stack_limit`. This assertion is correct since after parsing the rhs of the assignment and emitting `CBC_ASSIGN_PROP_THIS_LITERAL` the maximum actual stack depth is 1 or 2, depending on whether the class has computed initializer field(s). However we do not exact stack limit currently, but if the execution reaches this point it means the limit is less than `PARSER_MAXIMUM_STACK_LIMIT`.

So this could only cause problem if (in a really weird scenario) the stack limit would be `2` or `1` since increasing it to`3` or `2` would reach the limit. However in this case this line is not executed since the engine would already thrown error during the rhs part parsing of the assignment.

Reply:
I'm a bit hesitant. Let me explain. What would be the "real" problem? Is it stack_depth being larger than 2, or stack_limit growing above the max allowed value? After thorough analysis (your explanation above) we "know" (suspect, assume, expect) that stack_limit cannot grow too large, because rhs parsing would already throw error, etc. But that's kind of a "transitive" knowledge. If any (hypothetical) change happens in that part of the code, we have to revisit the assumptions here whether they still hold. Whether we can still rely on them here. Whether `stack_depth <= 2` still ensures that `stack_limit <= PARSER_MAXIMUM_STACK_LIMIT`. However, asserting directly on `stack_limit <= PARSER_MAXIMUM_STACK_LIMIT` does not have this problem. (I think)

Reply:
Fine, to reduce the transitive dependency, I've added the requested assertion.